pennylane/qaoa/cycle.py

```diff
@@ -489,5 +495,6 @@ def _inner_net_flow_constraint_hamiltonian(graph: nx.DiGraph, node) -> Hamiltoni
     coeffs, ops = _square_hamiltonian_terms(coeffs, ops)
     H = Hamiltonian(coeffs, ops)
     H.simplify()
-
+    # store the valuable information that all observables are in one commuting group
+    H.grouping_indices = [list(range(len(H.ops)))]
```

Comment:
I must admit, I am shocked how many times this needs to be declared 😆

Reply:
But in a sense the functions are there to construct Hamiltonians, and if this is info for the Hamiltonian it needs to be there?